### PR TITLE
Explore old ERC20 migration

### LINKED ISSUE #118: Explore guides to onboard legacy tokens in a sample project

### PR DESCRIPTION
**Pull request by facuspagnuolo**
Closes #118 

This PR adds a new example to the current opt-in strategy to onboard legacy tokens, ensuring old versions of the ERC20 provided in OZ@1.1.0 can be migrated to the latest provided version.

### References
- [Onboarding legacy token opt-in stgy](https://github.com/zeppelinos/labs/tree/master/migrating_legacy_token_opt_in#migrating-legacy-non-upgradeable-token-to-upgradeability-with-opt-in-strategy)
- [Old ERC20 issue with new versions of solc](https://gist.github.com/decanus/bdc08df5bf73af349333d1a8cae07d23)
- [Possible solution proposed by Brendan Chou](https://gist.github.com/BrendanChou/88a2eeb80947ff00bcf58ffdafeaeb61)

### CODE REVIEW

File: migrating_legacy_token_opt_in/contracts/old_erc20_token/MyToken_V1.sol [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  function migrateToken(uint _amount) public {
+    migrateTokenTo(_amount, msg.sender);
+  }
+
+  function migrateTokenTo(uint _amount, address _to) public {
+    legacyToken.transferFrom(msg.sender, burnAddress, _amount);
+    this.mint(_to, _amount);
+  }
+
+  // this methods are just for testing purposes
```
Comment (spalladino) [inline migrating_legacy_token_opt_in/contracts/old_erc20_token/MyToken_V1.sol]: this => these

File: migrating_legacy_token_opt_in/contracts/old_erc20_token/MyToken_V0.sol [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * @title MyToken V0
+ */
+contract MyToken_V0 is PausableToken, MintableToken {
+  using SafeMath for uint256;
+
+  string public name = "MyToken";
+  string public symbol = "MTK";
+  uint public decimals = 18;
+
+  function mintTimelocked(address _to, uint256 _amount, uint256 _releaseTime) onlyOwner canMint returns (TokenTimelock) {
```
Comment (spalladino) [inline migrating_legacy_token_opt_in/contracts/old_erc20_token/MyToken_V0.sol]: Why is this function here?

File: migrating_legacy_token_opt_in/contracts/old_erc20_token/MyToken_V1.sol [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    legacyToken = OldERC20(_legacyToken);
+  }
+
+  function migrateToken(uint _amount) public {
+    migrateTokenTo(_amount, msg.sender);
+  }
+
+  function migrateTokenTo(uint _amount, address _to) public {
+    legacyToken.transferFrom(msg.sender, burnAddress, _amount);
+    this.mint(_to, _amount);
```
Comment (spalladino) [inline migrating_legacy_token_opt_in/contracts/old_erc20_token/MyToken_V1.sol]: Should we change this to `require(this.mint(_to, _amount));`?

File: migrating_legacy_token_opt_in/contracts/old_erc20_token/MyToken_V0.sol (hunk 2 above)
Comment (facuspagnuolo) [reply migrating_legacy_token_opt_in/contracts/old_erc20_token/MyToken_V0.sol]: I was just mimicking  a real sample object

File: migrating_legacy_token_opt_in/contracts/old_erc20_token/MyToken_V1.sol (hunk 3 above)
Comment (facuspagnuolo) [reply migrating_legacy_token_opt_in/contracts/old_erc20_token/MyToken_V1.sol]: sure

**Pull request merged**